library/src/layout/par.rs

```diff
@@ -864,9 +875,29 @@ fn linebreak_optimized<'a>(vt: &Vt, p: &'a Preparation<'a>, width: Abs) -> Vec<L
             // Determine how much the line's spaces would need to be stretched
             // to make it the desired width.
             let delta = width - attempt.width;
-            let mut ratio = delta / attempt.stretch();
+            // Determine how much stretch are permitted.
+            let stretch = if delta >= Abs::zero() {
```

Comment:
its a bit confusing that "stretch" is used for both stretchability and shrinkability. maybe "adjust" or "adjustability"?

Reply:
Fixed in 8848269d3b1acb7c6fd9816331e8c980faa33c4f